sys/arm64/arm64/support.S

```diff
@@ -38,16 +38,26 @@ __FBSDID("$FreeBSD$");
 
 #include "assym.inc"
 
+#ifdef __CHERI_PURE_CAPABILITY__
+#error TODO Port to purecap
+#endif
+
 .macro check_user_access user_arg, limit, bad_addr_func
 	ldr	x7, =(\limit)
+#if __has_feature(capabilities)
+	gcvalue	x6, c\user_arg
+	cmp	x6, x7
+#else
 	cmp	x\user_arg, x7
+#endif
 	b.cs	\bad_addr_func
 .endm
 
 /*
  * One of the fu* or su* functions failed, return -1.
  */
 ENTRY(fsu_fault)
+	EXIT_C64
```

Comment:
This shouldn't be needed any more.

Reply:
Agreed, that got dropped on morello-dev but not here.